kairos-cli/src/crypto/signer.rs

```diff
@@ -0,0 +1,38 @@
+use std::path::Path;
+
+use super::private_key::CasperPrivateKey;
+use super::public_key::CasperPublicKey;
+use crate::crypto::error::CryptoError;
+use casper_types::bytesrepr::ToBytes;
+use casper_types::{crypto, ErrorExt, PublicKey};
+
+pub struct CasperSigner {
+    secret_key: CasperPrivateKey,
+    pub public_key: CasperPublicKey,
+}
+
+#[allow(unused)]
+impl CasperSigner {
+    pub fn from_file<P: AsRef<Path>>(file_path: P) -> Result<Self, ErrorExt> {
+        let secret_key = CasperPrivateKey::from_file(file_path)?;
+
+        // Derive the public key.
+        let public_key = CasperPublicKey(PublicKey::from(&secret_key.0));
```

Comment:
I would have considered separating file IO from retrieval of public key. I'd like granular control over error whether `from_file` fails because the file isn't there or because it's ill-formatted (which presumably would make `PublicKey::from` fail), and separating it into more smaller functions would be one way to do this.

Reply:
1. `PublicKey::from` cannot fail.

2. `CasperSigner::from_file` already provides granular errors, for example:

- non existing file :arrow_right: `ErrorExt::SecretKeyLoad(ReadFileError { ... })`
- invalid PEM content :arrow_right: `ErrorExt::FromPem("malformedframing")`